cpu/nrf51822/lpm_arch.c

```diff
@@ -27,8 +29,25 @@ void lpm_arch_init(void)
 
 enum lpm_mode lpm_arch_set(enum lpm_mode target)
 {
-    (void) target;
-    /* TODO: needs to be implemented */
+    switch (target) {
+        /* wait for next interrupt */
+        case LPM_IDLE:
+        case LPM_SLEEP:
+        case LPM_POWERDOWN:
+            __WFI();
```

Comment:
I think __DSB() is not supported on Cortex-M0. 

Here are power some down examples https://github.com/NordicSemiconductor/nrf51-powerdown-examples

My plan is to allow switching of the high frequency timer when possible. But this needs a lot more changes in RIOT code. 

Run current Timer 470uA, run current RTC 0.4uA (XTAL) or 15uA (RC).